shared/reducers/chat2.tsx

```diff
@@ -541,17 +541,25 @@ export default (_state: Types.State = initialState, action: Actions): Types.Stat
       }
       case Chat2Gen.unfurlTogglePrompt: {
         const {show, domain, conversationIDKey, messageID} = action.payload
-        const unfurlPromptMap = new Map(draftState.unfurlPromptMap || [])
-        const mmap = new Map(unfurlPromptMap.get(conversationIDKey) || [])
-        const prompts = new Set(mmap.get(messageID) || [])
+        const {unfurlPromptMap} = draftState
```

Comment:
i chose this as a simple example since it exposes updating maps and sets in one small isolated place
there are a couple of strategies for this kind of flow
1. (this) Make any missing maps / sets along the way and update it. Very vanilla js
1. Do a get/set, aka `const unformPromptMap = draftState..unfurmPromptMap || new Map()` (note note wrapping that whole thing in a map since we don't need to copy always now)
1. Avoid nesting maps of maps of sets etc (likely better)